tools/splitfasta/split_fasta.py

```diff
@@ -1,32 +1,41 @@
 #!/usr/bin/env python
 
+import argparse
 import os
-import sys
 
 from Bio import SeqIO
 
-num_chunks = 0
-if len(sys.argv) == 3:
-    num_chunks = int(sys.argv[2])
-    input_filename = sys.argv[1]
-elif len(sys.argv) == 2:
-    input_filename = sys.argv[1]
-else:
-    exit("Usage: split_fasta.py <input_filename> [<num_chunks>]")
+parser = argparse.ArgumentParser()
+parser.add_argument("--records", type=int, default=None)
+parser.add_argument("--limit", type=int, default=None)
+parser.add_argument("--num-chunks", type=int, default=0)
+parser.add_argument("input_file")
+args = parser.parse_args()
+
+input_filename = args.input_file
+num_chunks = args.num_chunks
+record_count = args.records
+record_limit = args.limit
 
 os.mkdir("splits")
 
-if num_chunks != 0:
-    # if splitting into chunks we need to count how many records are in the
-    # input file
+if record_limit and num_chunks > record_limit:
+    exit(f"ERROR: Requested number of chunks {num_chunks} exceeds limit {record_limit}")
+
+if not record_count and (num_chunks != 0 or record_limit):
+    # if no count is provided and if splitting into chunks or a limit is set, we need to count how many records are in the input file
     record_count = 0
     with open(input_filename) as input_file:
         for line in input_file:
             if line.lstrip().startswith(">"):
                 record_count += 1
 
+if num_chunks != 0:
     records_per_chunk = round(float(record_count) / num_chunks)
 
+if record_limit and record_count > record_limit:
+    exit(f"ERROR: Number of sequences {record_count} exceeds limit {record_limit}")
```

Comment:
@natefoo @bgruening isn't this wrong in the case of chunks?

Reply:
The chunk count is checked above on line 22 I believe?